src/client/Microsoft.Identity.Client/ManagedIdentity/ImdsV2ManagedIdentitySource.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class ImdsV2ManagedIdentitySource : AbstractManagedIdentity
+    {
+        /// <summary>
+        /// Factory method to create an instance of `CredentialManagedIdentitySource`.
```

Comment:
This factory method doesnt seem to have much logic in it besides the log message. Can the constructor be used to instantiate this instead? it returns a AbstractManagedIdentity but it looks like the type will always be ImdsV2ManagedIdentitySource anyway?

Reply:
leaving this as is for consistency @trwalke